File: 01-week-1--starter-algorithms/02-day-3--fibonacci-series/ruby/fibonacci.rb
def fibonacci(num)
  if num <= 1
    return num
  end

  a = 0
  b = 1
  i = 2

  until i > num do
    c = a + b
    a = b
    b = c
    i = i + 1
  end

  b
end

if __FILE__ == $PROGRAM_NAME
  puts "Expecting: 0"
  puts "=>", fibonacci(0)

  puts

  puts "Expecting: 1"
  puts "=>", fibonacci(2)

  puts

  puts "Expecting: 55"
  puts "=>", fibonacci(10)

  # Don't forget to add your own!
end

# Please add your pseudocode to this file
# And a written explanation of your solution

File: 17-week-13/00-day-1-to-2--dynamic-programming/javascript/fibonacci.js
let fibs = [];

function fibonacci(num) {
  if (num === 0 || num === 1) return num;
  if (fibs[num]) return fibs[num];
  fibs[num] = fibonacci(num - 1) + fibonacci(num - 2);
  return fibs[num];
}

if (require.main === module) {
  // add your own tests in here
  console.log("Expecting: 0");
  console.log("=>", fibonacci(0));

  console.log("");

  console.log("Expecting: 1");
  console.log("=>", fibonacci(2));

  console.log("");

  console.log("Expecting: 55");
  console.log("=>", fibonacci(10));
}

module.exports = fibonacci;

// Please add your pseudocode to this file
// And a written explanation of your solution
